openprescribing/dmd2/views.py

```diff
@@ -229,21 +235,45 @@ def bnf_code_relationships_view(request, bnf_code):
 
 
 def search_view(request):
-    q = request.GET.get("q")
+    if 'q' in request.GET:
+        form = SearchForm(request.GET)
 
-    ctx = {"q": q}
+        if form.is_valid():
+            search_params = form.cleaned_data
+            max_results_per_obj_type = search_params.pop('max_results_per_obj_type') or 10
+            results = search(**search_params)
+            _annotate_search_results(results, search_params, max_results_per_obj_type)
 
-    if q:
-        results = search(q)
+            if len(results) == 1:
```

Comment:
Can we have a comment here?  We could avoid a few moments of brow-furrowing in the future :)